nemo/collections/asr/models/sortformer_diar_models.py

```diff
@@ -571,10 +662,46 @@ def test_batch(
         logging.info(f"Batch Recall MEAN: {torch.mean(torch.tensor(self.batch_recall_list))}")
         logging.info(f"Batch ATS F1Acc. MEAN: {torch.mean(torch.tensor(self.batch_f1_accs_ats_list))}")
 
+    def on_validation_epoch_end(self) -> Optional[dict[str, dict[str, torch.Tensor]]]:
+        """Run validation with sync_dist=True."""
+        return super().on_validation_epoch_end(sync_metrics=True)
+
+    @torch.no_grad()
     def diarize(
         self,
-    ):
-        """One-clieck runner function for diarization."""
-        # TODO: A direct one-click runner function that generates
-        # speaker labels from audio file path lists.
-        raise NotImplementedError
+        audio: Union[str, List[str], np.ndarray, DataLoader],
+        batch_size: int = 1,
+        include_tensor_outputs: bool = False,
+        postprocessing_yaml: Optional[str] = None,
+        num_workers: int = 0,
+        verbose: bool = True,
+        override_config: Optional[DiarizeConfig] = None,
+    ) -> Union[List[List[str]], Tuple[List[List[str]], List[torch.Tensor]]]:
+        """One-click runner function for diarization.
+
+        Args:
+            audio: (a single or list) of paths to audio files or path to a manifest file.
+            batch_size: (int) Batch size to use during inference.
+                Bigger will result in better throughput performance but would use more memory.
+            include_tensor_outputs: (bool) Include raw speaker activity probabilities to the output.
```

Comment:
The name seems to suggest that the only diff is output type, but from the docstring it seems we also include VAD output when `include_tensor_outputs=True`. Do you think we can make the param name more informative?

Reply:
This was discussed to be kept and will be discussed later whether to be changed.